drink/src/session.rs

```diff
@@ -395,3 +470,71 @@ impl<R: Runtime> Session<R> {
         self.sandbox.override_debug_handle(d);
     }
 }
+
+/// A struct representing the result of parsing a `.contract` bundle file.
+///
+/// It can be used with the following methods of the `Session` struct:
+/// - `deploy_bundle`
+/// - `deploy_bundle_and`
+/// - `upload_bundle`
+/// - `upload_bundle_and`
+pub struct ContractBundle {
+    /// WASM blob of the contract
+    bytes: Vec<u8>,
+    /// Transcoder derived from the ABI/metadata
+    transcoder: Rc<ContractMessageTranscoder>,
+}
+
+impl ContractBundle {
+    /// Load and parse the information in a `.contract` bundle under `path`, producing a `ContractBundle` struct.
+    pub fn load<P>(path: P) -> Result<Self, SessionError>
+    where
+        P: AsRef<std::path::Path>,
+    {
+        let metadata: ContractMetadata = ContractMetadata::load(&path).map_err(|e| {
+            SessionError::BundleLoadFailed(format!("Failed to load the contract file:\n{e:?}"))
+        })?;
+
+        let ink_metadata = serde_json::from_value(serde_json::Value::Object(metadata.abi))
+            .map_err(|e| {
+                SessionError::BundleLoadFailed(format!(
+                    "Failed to parse metadata from the contract file:\n{e:?}"
+                ))
+            })?;
+
+        let bytes = metadata
+            .source
+            .wasm
+            .ok_or(SessionError::BundleLoadFailed(
+                "Failed to get the WASM blob from the contract file".to_string(),
+            ))?
+            .0;
+
+        let transcoder = Rc::new(ContractMessageTranscoder::new(ink_metadata));
+
+        Ok(Self { bytes, transcoder })
+    }
+
+    /// Load the `.contract` bundle (`bundle_file`) located in the `project_dir`` working directory.
+    ///
+    /// This is meant to be used predominantly by the `local_bundle!` macro.
+    pub fn local(project_dir: &str, bundle_file: String) -> Self {
+        let mut path = PathBuf::from(project_dir);
+        path.push("target");
+        path.push("ink");
+        path.push(bundle_file);
+        Self::load(path).expect("Loading the local bundle failed")
+    }
+}
+
+/// A convenience macro that allows you to load a bundle found in the target directory
+/// of the current project.
+#[macro_export]
+macro_rules! local_bundle {
+    () => {
+        drink::session::ContractBundle::local(
+            env!("CARGO_MANIFEST_DIR"),
+            env!("CARGO_CRATE_NAME").to_owned() + ".contract",
+        )
+    };
+}
```

Comment:
bc you want it to be resolved in the target crate, not in drink :P

Reply:
but if it's used as a function in the body of the contract's test then it will resolve there, right?

Reply:
Aye! That might be it.

Reply:
Right, so to the best of my knowledge (albeit very empirical) you have three options for loading the env vars in library code:
* `env!` inside a regular function: during the compilation of the library crate, in which case `CARGO_CRATE_NAME` will be `drink` -- not helpful
* `std::env::var`: loading at runtime. `CARGO_CRATE_NAME` won't be defined because it's set only for the duration of the compilation
* `env!` inside a macro, at which point it will be substituted during the compilation of the project that's using said library crate -- bingo :)

If I'm missing something let me know!